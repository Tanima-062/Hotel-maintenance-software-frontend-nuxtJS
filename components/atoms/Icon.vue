<template>
  <div :class="styles">
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      :viewBox="`0 0 24 24`"
    >
      <path :d="icon" />
    </svg>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  get styles(): string {
    const classList = []
    if (this.bgColor) {
      classList.push(`${this.bgColor}`)
    }
    if (this.round) {
      classList.push('rounded-full')
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }
}
</script>
