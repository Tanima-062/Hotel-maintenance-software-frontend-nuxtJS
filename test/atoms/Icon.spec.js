import { shallowMount } from '@vue/test-utils'
import Icon from '@/components/atoms/Icon.vue'

describe('Atoms Icon Check', () => {
  test('Icon Test', () => {
    const props = {
      icon: 'mdiClose',
      color: 'text-black-dark',
      round: true,
    }
    const wrapper = shallowMount(Icon, {
      propsData: props,
    })

    // check props value
    expect(wrapper.props().icon).toBe(props.icon)

    // check if it has a class
    expect(wrapper.find('svg').classes()).toContain(props.color)

    // check rounded
    if (props.round) {
      expect(wrapper.classes()).toContain('rounded-full')
    }

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
