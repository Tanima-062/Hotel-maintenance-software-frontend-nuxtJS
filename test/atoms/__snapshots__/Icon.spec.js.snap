// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Icon Check Icon Test 1`] = `
<div
  class="rounded-full"
>
  <svg
    class="w-7 h-7 fill-current text-black-dark"
    style="width: 24px; height: 24px;"
    viewBox="0 0 24 24"
  >
    <path
      d="mdiClose"
    />
  </svg>
</div>
`;
